src/System.Data.SqlClient/src/System/Data/SqlClient/SqlConnection.cs

```diff
@@ -462,8 +462,28 @@ private void CloseInnerConnection()
 
         override public void Close()
         {
+            ConnectionState previousState = State;
+            Guid operationId;
+            Guid clientConnectionId;
+
+            // during the call to Dispose() there is a redundant call to 
```

Comment:
That's a good point, and yes it is actually a redundant call (stepped through debugger after odd test results). I agree that your approach would seem the best, but that could have unforseen regression issues, and that'd be a better question for the SqlClient team. @saurabh500 what are your thoughts on this?